tuf/signed/verify.go

```diff
@@ -107,3 +108,18 @@ func VerifySignature(msg []byte, sig *data.Signature, pk data.PublicKey) error {
 	sig.IsValid = true
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks if the private key and the public keys forms valid key pairs.
+// This should work with both ecdsa-x509 certificate PublicKey as well as ecdsa PublicKey
+func VerifyPublicKeyMatchesPrivateKey(privKey data.PrivateKey, pubKey data.PublicKey) error {
+	pubKeyID, err := utils.CanonicalKeyID(pubKey)
+	privKeyID := privKey.ID()
```

Comment:
Actually apologies it looks like `utils.CanonicalKeyID` also assumes that the key is not nil, and will also panic :|  If you wouldn't mind fixing it in this PR, that would be awesome but I'd also be happy to fix it in a separate PR.